front-end/src/app/shared/models/transaction-types/JF_TRANSFER_NATIONAL_PARTY_HEADQUARTERS_ACCOUNT.model.ts

```diff
@@ -0,0 +1,47 @@
+import { LabelUtils } from 'app/shared/utils/label.utils';
+import { schema } from 'fecfile-validate/fecfile_validate_js/dist/JF_TRANSFER_NATIONAL_PARTY_HEADQUARTERS_ACCOUNT';
+import { TransactionType } from '../../interfaces/transaction-type.interface';
+import {
+  AggregationGroups,
+  SchATransaction,
+  ScheduleATransactionTypeLabels,
+  ScheduleATransactionTypes,
+} from '../scha-transaction.model';
+import {
+  CANCEL_CONTROL,
+  SAVE_ANOTHER_CONTROL,
+  SAVE_LIST_CONTROL,
+  TransactionNavigationControls,
+} from '../transaction-navigation-controls.model';
+
+export class JF_TRANSFER_NATIONAL_PARTY_HEADQUARTERS_ACCOUNT implements TransactionType {
+  scheduleId = 'A';
+  componentGroupId = 'E';
+  isDependentChild = false;
+  title = LabelUtils.get(ScheduleATransactionTypeLabels, ScheduleATransactionTypes.JF_TRANSFER_NATIONAL_PARTY_HEADQUARTERS_ACCOUNT);
+  schema = schema;
+  transaction?: SchATransaction;
+  subTransactionTypes = [
+    ScheduleATransactionTypes.INDIVIDUAL_NATIONAL_PARTY_HEADQUARTERS_JF_TRANSFER_MEMO,
+    ScheduleATransactionTypes.PAC_NATIONAL_PARTY_HEADQUARTERS_JF_TRANSFER_MEMO,
+    ScheduleATransactionTypes.TRIBAL_NATIONAL_PARTY_HEADQUARTERS_JF_TRANSFER_MEMO,
+    ScheduleATransactionTypes.PARTNERSHIP_NATIONAL_PARTY_HEADQUARTERS_JF_TRANSFER_MEMO,
+  ];
+  navigationControls: TransactionNavigationControls = new TransactionNavigationControls(
+    [],
+    [CANCEL_CONTROL],
+    [SAVE_LIST_CONTROL, SAVE_ANOTHER_CONTROL]
+  );
+
+  contributionPurposeDescripReadonly(): string {
+    return 'Headquarters Buildings Account Transfer of JF Proceeds';
+  }
+
+  getNewTransaction() {
+    return SchATransaction.fromJSON({
+      form_type: 'SA17',
+      transaction_type_identifier: ScheduleATransactionTypes.JF_TRANSFER_NATIONAL_PARTY_HEADQUARTERS_ACCOUNT,
+      aggregation_group: AggregationGroups.NPARTY_HEADQUARTERS,
```

Comment:
you'll want to pull develop.  The AggregationGroups have been updated

Reply:
aggregation group was updated